liquibase-standard/src/main/java/liquibase/change/ChangeFactory.java

```diff
@@ -120,6 +134,43 @@ public Change create(String name) {
         }
     }
 
+    /**
+     * Verify if the supports method is implemented in the change if it's not part of the default liquibase changes
+     */
+    private void verifySupportsMethodImplementation(Set<Change> plugins) {
+        if (GlobalConfiguration.SUPPORTS_METHOD_VALIDATION_LEVELS.getCurrentValue().equals(SupportsMethodValidationLevelsEnum.OFF)) {
+            return;
+        }
+        //we only verify supports method if this is not part of the default liquibase changes
+        for (Change plugin : plugins) {
+            String packageName = plugin.getClass().getPackage().getName();
+            if (packageName.startsWith("liquibase.change") || packageName.startsWith("com.datical.liquibase.ext")) {
+                continue;
+            }
+
+            try {
+                // if the supports method is not implemented in the plugin show the wraning according to the defined level
+                if (plugin.getClass().getMethod("supports", Database.class).getDeclaringClass().getPackage().getName().startsWith("liquibase.change")) {
+                    if (LiquibaseUtil.getBuildVersion().equals(LiquibaseUtil.DEV_VERSION)) {
+                        throw new UnexpectedLiquibaseException(String.format(SUPPORTS_METHOD_REQUIRED_MESSAGE, plugin.getClass().getName()));
+                    }
+                    switch (GlobalConfiguration.SUPPORTS_METHOD_VALIDATION_LEVELS.getCurrentValue()) {
+                        case WARN:
+                            Scope.getCurrentScope().getLog(getClass()).warning(String.format(SUPPORTS_METHOD_REQUIRED_MESSAGE, plugin.getClass().getName()));
+                            plugins.remove(plugin);
+                            break;
+                        case FAIL:
+                            throw new UnexpectedLiquibaseException(String.format(SUPPORTS_METHOD_REQUIRED_MESSAGE, plugin.getClass().getName()));
+                        default:
+                            break;
+                    }
+                }
+            } catch (NoSuchMethodException e) {
+                throw new UnexpectedLiquibaseException(String.format(SUPPORTS_METHOD_REQUIRED_MESSAGE, plugin.getClass().getName()));
```

Comment:
Why are you throwing an exception from here?

Reply:
This part of the code will never be reached as the "supports" method will always be there as it is defined in the interface - but reflection throws the checked exception so I just added this extra throws.